torch/utils/cpp_extension.py

```diff
@@ -1350,7 +1350,11 @@ def _get_build_directory(name, verbose):
         if verbose:
             print('Creating extension directory {}...'.format(build_directory))
         # This is like mkdir -p, i.e. will also create parent directories.
-        os.makedirs(build_directory)
+        try:
```

Comment:
here's the actual change

Reply:
why did you have to do this? what was the case for 777?

Reply:
It's explained in https://github.com/pytorch/pytorch/issues/34238 but I agree with Sam, using a cache directory is a better plan

Reply:
i'll update the PR to use a cache dir.